### ISSUE TITLE
Set page size for the output

### ISSUE THREAD

**Issue by bytrangle** [cfcurtis/pdfstitcher#217]
As I understand, this tool is primarily used for projecting. For projecting, the final size of the output is irrelevant.

However, I think this tool is also awesome for assembling an A0 page from multiple A4 pages. It's faster, easier to use, and more accurate than any other methods I've known. However, there is nothing in the tool that allows users to set the output's dimension.

Are you open to a feature for configuring output's dimension? My main language is Javascript, but I can take the time to help.

**Comment by cfcurtis** [comment]
Hi, thanks for the suggestion and the offer! I've been asked about this kind of thing before, but the issue is that most sewing patterns don't tile nicely to an A0 size. Would you want to tile it and then split into multiple A0 pages?

There is actually a target size option that scales the output to fit a given page dimension, but it's only exposed on the command line as it's not appropriate for sewing patterns.

**Comment by bytrangle** [comment]
I've realized that I've been using the "tile pages" feature. I suppose we'll need to calculate the how many A0 pages needed based on the number of columns (or rows) specified by the user.
What is the command for running Pdf Stitcher in CLI?

**Comment by cfcurtis** [comment]
Yes, the out of bounds thing is what I meant by not tiling nicely to an A0 size. Would people still want to create A0 pages if they need to be cut and taped anyway?

The CLI is poorly documented at the moment (and needs improvements, particularly for the layer stuff), but if you've installed via pip then you can run the command `pdfstitcher --help` to see the various options. For example, to tile all pages of a document with 5 columns, the command would be:

```bash
pdfstitcher --columns 5 infile.pdf outfile.pdf
```

To scale to fit, you can specify a target height and width e.g.:

```bash
pdfstitcher -c 5 --target-width 33.1 --target-height 46.8 infile.pdf outfile.pdf
```

Both height and width need to be defined right now, and it will crash if the file doesn't have layers. I pushed a fix for this in the `cli_improvements` branch, but haven't put out a new release yet.

**Comment by bytrangle** [comment]
> Yes, the out of bounds thing is what I meant by not tiling nicely to an A0 size. Would people still want to create A0 pages if they need to be cut and taped anyway?

I'm afraid I don't follow this. Suppose each A0 page can take 5 columns and 3 rows of portrai A4, I can fit at most 15 A4 on each A0 page. If the original A4 pattern has 40 pages, I will need 4 A0 pages to tile all of the A4.

People can send this output pdf of size A0 with 4 pages to the copy shop to print. Why does one need to cut and tape?

**Comment by cfcurtis** [comment]
> People can send this output pdf of size A0 with 4 pages to the copy shop to print. Why does one need to cut and tape?

Because the boundaries of those pages will still cut through pattern pieces, unless the designer laid them out to fit A0 to begin with (in which case, stitching is not needed). It's still a lot less work to assemble 4 pages than 40 though so it could be something people will want.

> On a second thought, how do I know that each A0 takes 5 columns and 3 rows 🤔? I laid them out on Inkscape. So unless people have a similar tool to play around the A4, it can be hard to visualize how they all fit.

Fitting the pages into one or several A0s can be calculated automatically, but finding the best layout for the pattern pieces definitely needs some visuals. This is why I haven't bothered to implement such a feature in the past :)

**Comment by gaby64** [comment]
I implemented changes to achieve my desired effect, unrelated to stitching btw, I wanted it to collapse the free space on shipping labels

https://github.com/gaby64/pdfstitcher/commit/7ed3b4edb024c7834237680c228191e2887df020
